### ISSUE TITLE
bad argument #1 to '?' 

### ISSUE THREAD

**Issue by noirblack** [kimiyoung/review_net#3]
Thanks for sharing the code, it' s a wonderful job.
I have extracted the features and save them in data/ 
Howerer, I got an error with the following command :
**th main.lua -model_pack reason_att_copy.lua -save_file -save_file_name model/ra.model**
[](url)
cloning 30                                                             
cloning soft_att_lstm                                                  
cloning reasoning lstm                                                 
/home/julius/torch/install/bin/luajit: bad argument #1 to '?' (string e
xpected, got nil)                                                      
stack traceback:                                                       
        [C]: at 0x7f054d482ad0                                         
        [C]: in function 'DiskFile'                                    
        /home/julius/torch/install/share/lua/5.1/torch/File.lua:405: in
 function 'load'                                                       
        ./dataloader.lua:127: in function 'gen_train_data'             
        ...ect/review_net/image_caption_offline/reason_att_copy.lua:356
: in function 'train'                                                  
        main.lua:33: in main chunk                                     
        [C]: in function 'dofile'                                      
        ...lius/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145
: in main chunk                                                        
        [C]: at 0x00406670                                             
[](url)

**Comment by kimiyoung** [comment]
This is because the corresponding feature file does not exist for the given image id.

You can check the following folders:
data/train2014_features_vgg_vd19_fc7
data/val2014_features_vgg_vd19_fc7
to see if everything looks normal.

Also you can print out the image id (right before line 127 in dataloader.lua) and check whether the corresponding feature file exists in the above folders.

**Comment by noirblack** [comment]
@kimiyoung Thanks a lot! 
Based on your comments, I find that some file is missing in data/val2014_features_vgg_vd_fc7/.
I extract the features again and the training is going well.

**Comment by getengqing** [comment]
Hello, when I run the "th main.lua -save_file_name models/reason_att_copy_simp_seed13.model -seed 13 -server_train_mode", I got the error:
Initialize dataloader...	
/home/torch/install/bin/luajit: ./utils/anno_utils_filter.lua:156: attempt to index a nil value
stack traceback:
	./utils/anno_utils_filter.lua:156: in function 'read_captions'
	./dataloader.lua:41: in function '__init'
	/home/getengqing/torch/install/share/lua/5.1/torch/init.lua:91: in function </home/torch/install/share/lua/5.1/torch/init.lua:87>
	[C]: in function 'DataLoader'
	main.lua:21: in main chunk
	[C]: in function 'dofile'
	...qing/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
	[C]: at 0x00406670
How I solve it?Ths!
